src/text/parse_result.rs

```diff
@@ -51,7 +51,7 @@ pub(crate) type IonParseResult<'a, O> = IResult<&'a str, O, IonParseError<'a>>;
 /// with other parsers that return `IonParseResult`, you can call `upgrade()` on your parser's
 /// output before returning it.
 ///
-/// ```ignore
+/// ```ignore // Doc tests cannot use internal APIs
```

Comment:
🗺️ This doc comment's code fence _does_ contain Rust code, so `ignore` is appropriate, but it also uses crate-visible methods and so will not compile.

Reply:
It seems like a bad idea to have doc comment code that uses private/pub(crate) functions. Aren't you basically publishing an example that the reader can never use?

Reply:
This is a private module, so it isn't exposed in the user-facing documentation.

Any doc comment that includes a code fence is treated as code that should compile using the public API; I'm unsure how to convince `cargo` to compile it with pub(crate) APIs. If you know of a way, let me know!